--- libnids-1.16/configure.in.orig	Tue Oct 28 12:51:56 2003
+++ libnids-1.16/configure.in	Tue Oct 28 12:54:26 2003
@@ -35,8 +35,8 @@
 dnl Checks for library functions.
 AC_CHECK_FUNCS(gettimeofday)
 dnl XXX - Solaris sux.
-AC_CHECK_LIB(socket, socket)
-AC_CHECK_LIB(nsl, gethostbyname)
+AC_SEARCH_LIBS(socket, socket)
+AC_SEARCH_LIBS(gethostbyname, nsl)
 
 dnl Checks for libpcap
 AC_MSG_CHECKING(for libpcap)
